massgov/mayflower
# React/DP-14085: Feedback form component.

# Description
This adds the FeedbackForm component for form submissions to Formstack. Form submissions use JSONP to communicate with the Formstack server in a similar manner as to the Drupal site (but without jQuery). There are checks in code to ensure `window` exists, so this should be alright to use with server side rendering.

Character Counting is being handled by the [react character counter](https://www.npmjs.com/package/react-character-counter) package.

JSONP is handled with the[ b-jsonp](https://github.com/brandonjpierce/b-jsonp) package.
I also updated the Paragraph component to finally support `props.children` so that you can do something like:
```
<Paragraph>My <strong>HTML</strong> here</Paragraph>
```

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-14085)

## Steps to Test

1. Pull the branch and run `npm install`. Run `npm run start` and browse to organisms/FeedbackForm.
2. Choose yes or no. If yes, the form should submit with or without anything typed into the textarea. If no, the textarea should be required and the form must only submit when there's text. 
3. Verify that either textarea's character counter decreases/increases as you type.
4. Verify that form submissions make it through to Formstack.

## Screenshots
<img width="1458" alt="Screen Shot 2019-05-30 at 5 45 52 PM" src="https://user-images.githubusercontent.com/5257660/58666992-d0564a00-8302-11e9-8381-8166471944a2.png">
